### PR TITLE
macOS compatibility

### PR DESCRIPTION
**Pull request by insidegui**
Related issue: #39 

Adding support for macOS was quite simple since the framework uses Core Animation which is the same between iOS and macOS :)

Here's what I have done:
- Created a compatibility header to import the correct frameworks when building on macOS vs iOS
- Added categories to `NSBezierPath`, `NSImage` and `NSValue` implementing some features used by keyframes which were not available on macOS
- Updated the Podspec to support macOS
- Added sample projects for macOS (Swift and Objective-C)

I've used the sample logo for testing, it looks exactly the same between iOS and macOS. Tested both manual integration and using CocoaPods, both work.

### CODE REVIEW

**Comment by insidegui** [comment]
FYI: I've also created [a little document-based app](https://github.com/insidegui/KeyframesPlayer) to be able to easily open `.json` animations for testing :)

File: ios/keyframes-sample-macos/objc/keyframes-sample-macos/AppDelegate.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//
+//  AppDelegate.h
+//  keyframes-sample-macos
+//
+//  Created by Guilherme Rambo on 12/12/16.
+//  Copyright © 2016 Guilherme Rambo. All rights reserved.
+//
+
```
Comment (LazyChild) [inline ios/keyframes-sample-macos/objc/keyframes-sample-macos/AppDelegate.h]: We need to use the same header as other sample apps

File: ios/keyframes-sample-macos/objc/keyframes-sample-macos/Info.plist [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleVersion</key>
+	<string>1</string>
+	<key>LSMinimumSystemVersion</key>
+	<string>$(MACOSX_DEPLOYMENT_TARGET)</string>
+	<key>NSHumanReadableCopyright</key>
+	<string>Copyright © 2016 Guilherme Rambo. All rights reserved.</string>
```
Comment (LazyChild) [inline ios/keyframes-sample-macos/objc/keyframes-sample-macos/Info.plist]: The copy right should be same as other sample app

File: ios/keyframes-sample-macos/swift/keyframes-sample-macos/Info.plist [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleVersion</key>
+	<string>1</string>
+	<key>LSMinimumSystemVersion</key>
+	<string>$(MACOSX_DEPLOYMENT_TARGET)</string>
+	<key>NSHumanReadableCopyright</key>
+	<string>Copyright © 2016 Guilherme Rambo. All rights reserved.</string>
```
Comment (LazyChild) [inline ios/keyframes-sample-macos/swift/keyframes-sample-macos/Info.plist]: ditto

File: ios/keyframes/src/Compatibility/Compatibility.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
+//
+//  Compatibility.h
+//  keyframes
+//
+//  Created by Guilherme Rambo on 12/12/16.
+//  Copyright © 2016 facebook. All rights reserved.
+//
+
```
Comment (LazyChild) [inline ios/keyframes/src/Compatibility/Compatibility.h]: We need use the same header as other code :)

**Comment by LazyChild** [comment]
Thank you so much @insidegui. This is really exciting!

- Headers should be updated
- All files under DataModel/ are auto-generated. In theory we shouldn't modify them, but I don't see an easy way to change the script for our compatibility usage. It might make sense to maintain the data model manually from now on.
- Also code need rebase

**Comment by insidegui** [comment]
I will implement the changes tomorrow and update the PR.

**Comment by insidegui** [comment]
Squashed and rebased :)

I've also implemented the framework improvements from #58 on the macOS target.

File: ios/keyframes-sample-macos/objc/keyframes-sample-macos/main.m [hunk 5]
```diff
@@ -0,0 +1,13 @@
+//
+//  main.m
+//  keyframes-sample-macos
+//
+//  Created by Guilherme Rambo on 12/12/16.
+//  Copyright © 2016 Guilherme Rambo. All rights reserved.
+//
```
Comment (LazyChild) [inline ios/keyframes-sample-macos/objc/keyframes-sample-macos/main.m]: missing one

File: ios/keyframes-sample-macos/swift/keyframes-sample-macos/keyframes-sample-macos-Bridging-Header.h [hunk 6]
```diff
@@ -0,0 +1,7 @@
+//
```
Comment (LazyChild) [inline ios/keyframes-sample-macos/swift/keyframes-sample-macos/keyframes-sample-macos-Bridging-Header.h]: Header is wrong.
And also the iOS swift sample is not using bridging header any more. Let's follow that pattern.

**Comment by LazyChild** [comment]
Thank you @insidegui. One last thing, objc sample app won't compile since we renamed the framework to capital K.

**Comment by skagedal** [comment]
If remodel is indeed not to be used any longer, may I suggest that the "auto-generated" comments are removed/changed? I was about to prepare a little pull request, installed remodel-gen and was surprised to see that it didn't generate the same files.

**Pull request merged**